### ISSUE TITLE
[NuGet.org Bug]: Download counters not updating for pre-release packages?

### ISSUE THREAD

**Issue by jodydonetti** [NuGet/NuGetGallery#8979]
### Impact

It bothers me. A fix would be nice

## TL/DR

It seems to me that the download counter is not working for versions of packages that are marked as pre-release.

The total download for a package that has pre-release versions but also "normal" versions seems to work fine, even though each pre-release version is stuck at 0 (zero).

Please note that the usual delay of 24/48h for the stats to be updated is already passed, since we are at around 6 days.
Also, in these 6 days the other downloads seem to have been updated just fine: I'm talking about my **global** downloads, the downloads for each **package as a whole**, and the downloads for each **package version** that is NOT a pre-release version.

## The setup

Up until last week I had 3 packages on nuget.org , all of them had some versions released in the last year, and none of those versions were pre-release (per semantic versioning).

All was fine.

Last week (Jan 27th to be precise) I uploaded 2 completely new packages (both with a pre-release version) and 3 new versions (again, all pre-release) of the 3 existing packages.
So all of these were in "alpha1" version.
Then about 3 days after that i pushed another pre-release version ("alpha2") for all of the 5 packages.

You can see all 5 of them [here](https://www.nuget.org/profiles/jodydonetti) (screenshot taken 2022-02-02):

![image](https://user-images.githubusercontent.com/1010086/152252639-1d57cb56-60c9-4d44-8e71-c5548df49e88.png)

(Now don't be fooled by the 0.1.9 you can see as the "Latest version" for the first 3 packages, since what is shown is the latest non-pre-release version, as I'm sure you know)

So, all these packages have an actual last version that is pre-release (currently "alpha2").

## The Problem

The problem is that since then, the download counters for each pre-releae version remained stuck at 0 (zero), even though me and other people already downloaded them.

See screenshots below for some evidence.

### Repro Steps

Hypothetically:
- upload a new pre-release version of an existing package OR upload a completely new package (with a pre-release version)
- make some downloads
- wait a couple of days
- download stats will not increase

### Expected Behavior

Download stats SHOULD increase.

## New package

Downloads for one of the new packages ([this](https://www.nuget.org/packages/ZiggyCreatures.FusionCache.Backplane.StackExchangeRedis/0.1.0-alpha2)) after 6 days:

![image](https://user-images.githubusercontent.com/1010086/152254941-84a73ae1-4146-495c-8c6a-8de0828ef696.png)

## Existing package

Downloads for one of the existing packages ([this](https://www.nuget.org/packages/ZiggyCreatures.FusionCache/0.1.9)) after 6 days:

![image](https://user-images.githubusercontent.com/1010086/152255046-23f4fcda-e4cc-4c13-957e-a64710c3129f.png)

Note that as, I said above, in the last 6 days the downloads for the v0.1.9 kept increasing.

**Comment by joelverhagen** [comment]
Hello @jodydonetti, apologies on the inconvenience. We have an ongoing issue with delayed package statistics. We have a notice on our [status page](https://status.nuget.org/).

> Package downloads are delayed since January 25th, 2022. Download count statistics are currently being ingested at a reduced rate. This is resulting in download counts on packages not reflecting their latest status. We apologize for the inconvenience and are actively working on a solution.

**Comment by jodydonetti** [comment]
Oh damn, sorry for the very long (and useless) issue!

The thing that tricked me was the fact that packages downloads were in fact increasing, just not for the pre-release versions which felt quite strange.

Honestly before posting I searched for some tweets about stats problems but was not able to find any, and I did not spot the status page link in the site footer, my bad.

Sorry for having wasted your time, and thanks for the support!

**Comment by jodydonetti** [comment]
Hi @joelverhagen I know the issue with delayed package stats is still goign on, but I just wanted to update you on the fact that after several days my download stats keep increasing for all my packages and package versions, **except** the pre-release one, which are still at a flat zero.
Maybe it's nothing, but since you are still investigating the issue and since the fact that only the pre-release ones are at zero is strange, maybe this could be something useful.
Thanks.

**Comment by joelverhagen** [comment]
@jodydonetti thanks for the additional data point. Historically prerelease versions are adopted at lower rates than stable versions so, without digging very deeply into our pipeline, I would guess that this effect you are seeing is more related to users' download behavior than this specific bug. That being said, it _is_ suspicious so let's leave this bug open until we've mitigated impact so we can verify downloads are indeed flowing properly for your package.

**Comment by jodydonetti** [comment]
> Historically prerelease versions are adopted at lower rates than stable versions so [...] I would guess that this effect you are seeing is more related to users' download behavior than this specific bug

I thought about the same (pre-releases are less picked up etc) but in 11 days since the release the download count is not 50 or 20 or even 10, it is absolutely flat zero for all the alpha versions:

![image](https://user-images.githubusercontent.com/1010086/152867662-82d484d5-a54d-4d47-a9c2-6bbc762a01db.png)

And I know for a fact that some people downloaded them, and even more so that I myself downloaded them the very first day for some projects I'm working on.

The other explanation is that, if since the 25th Jan (beginning of your current issue) you have still collected the download data - I certainly hope they are not lost - and are slowly processing them, maybe you are processing data between the 25th and the 28th (the day of my first alpha pre-release) so those number are not yet summed up.

Anyway, just my 2 cents: good luck with your issue resolution!

**Comment by joelverhagen** [comment]
>  I certainly hope they are not lost - and are slowly processing them, maybe you are processing data between the 25th and the 28th (the day of my first alpha pre-release) so those number are not yet summed up.

Nope, no lost data 👍

We're just slowly crunching through many gigabytes of logs! To provide a little more context, our old statistics pipeline is based on SQL and has had progressively worsening scale problems over the past several years. We're in the process of moving off of this older pipeline to something based on Azure Synapse but, unfortunately, the old pipeline ran into a serious problem before this cutover was complete. Not ideal 😃 but you don't have to be worried lost data and things should be more stable with the new system.

**Comment by jodydonetti** [comment]
> Nope, no lost data 👍

![image](https://user-images.githubusercontent.com/1010086/152869867-6216331f-645b-4793-912c-5499df9b823e.png)

> We're just slowly crunching through many gigabytes of logs! To provide a little more context [...]

**Comment by jodydonetti** [comment]
HI @joelverhagen , just wanted to update you.

![image](https://user-images.githubusercontent.com/1010086/153517837-c8c2991d-3078-4674-843d-4680e9cf17fb.png)

It seems some downloads are appearing for the `alpha1` versions, so the theory outlined seemed correct: it was still procesing the logs before my first alpha release (Jan 28th) and now it has reached that date. Also, `alpha2` and later `beta1` are still at 0, meaning it is recovering but has not yet reached these 2 dates (`alpha2` on Jan 31th and `beta1` on Feb 9th).

Something like this:

```
 |
25th Jan: start of problems on nuget.org
 |
 |
 |
28th Jan: my alpha1
 |
 *   <- it's probably arrived here
 |
31th Jan: my alpha2
 |
 |
 |
 |
 |
9th Jan: my beta1
 |
...
```

All in all I think we can close this issue, meaning that when the log crunching will be on par, the numbers will be all "normal".

Thanks!
